Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -8876,20 +8876,26 @@ bool WebPage::isUsingUISideCompositing() const
 void WebPage::setLinkDecorationFilteringData(Vector<WebCore::LinkDecorationFilteringData>&& strings)
 {
     m_linkDecorationFilteringData.clear();
-    m_domainScopedLinkDecorationFilteringData.clear();
 
     for (auto& data : strings) {
-        if (data.domain.isEmpty()) {
-            m_linkDecorationFilteringData.add(data.linkDecoration);
+        if (!m_linkDecorationFilteringData.isValidKey(data.linkDecoration)) {
+            ASSERT_NOT_REACHED();
```

Comment:
Maybe add a release log for this, too. We'll only crash with the assertion in a pretty specific case.